py3status/modules/usbguard.py

```diff
@@ -0,0 +1,350 @@
+                format_notification['action'] = action
+                format_notification = self.py3.safe_format(
+                    self.format_notification, format_notification
+                )
+                notification = self.py3.get_composite_string(format_notification)
+                self.py3.notify_user(
+                    notification,
+                    title='USBGuard',
+                    icon='/usr/share/icons/hicolor/scalable/apps/usbguard-icon.svg',
+                )
+
+            # apply policy
+            if 'usbguard_id' in self.data and index != 'sep':
+                self.proxy.applyDevicePolicy(
+                    self.data['usbguard_id'], targets[action], self.is_permanent
+                )
+
+    def _manipulate_all_devices(self, data):
+        format_all_devices = []
+        format_device_separator = self.py3.safe_format(self.format_device_separator)
+        self.py3.composite_update(format_device_separator, {'index': 'sep'})
+
+        for device in data:
+            for x in self.thresholds:
+                if x in data[device]:
+                    self.py3.threshold_get_color(data[device][x], x)
+
+            device_formatted = self.py3.safe_format(
+                self.format_all_devices, data[device]
+            )
+
+            # add usbguard_id
+            self.py3.composite_update(device_formatted, {'usbguard_id': data[device]['usbguard_id']})
+
+            format_all_devices.append(device_formatted)
+
+        for index, device in sorted(enumerate(format_all_devices), reverse=False):
+            format_all_devices[index] = device
+
+        format_all_devices = self.py3.composite_join(
+            format_device_separator, format_all_devices
+        )
+
+        return format_all_devices
+
+    def usbguard(self):
+        if self.error:
+            self.py3.error(str(self.error), self.py3.CACHE_FOREVER)
+
+        self.all_devices = self._get_all_devices()
+        self.usbguard_data['format_all_devices'] = self._manipulate_all_devices(
+            self.all_devices
+        )
+
+        response = {'cached_until': self.py3.CACHE_FOREVER, 'urgent': True}
+
+        if len(self.usbguard_data) > 0:
+            composite = self.py3.safe_format(self.format, self.usbguard_data)
+            response['composite'] = composite
+        else:
+            response['full_text'] = self.py3.safe_format(
+                self.format, {'format_all_devices': ''}
+            )
+
+        return response
+
+    def kill(self):
+        self.killed.set()
+
+    def on_click(self, event):
+        button = event['button']
+
+        self.py3.log('# click event')
+        self.py3.log(event)
+                
+        if button == self.button_allow:
+            self._set_policy('allow', index)
+        elif button == self.button_reject:
+            self._set_policy('reject', index)
+        elif button == self.button_block:
+            self._set_policy('block', index)
```

Comment:
You have three `index` here but no `index = event['index']`. Plz make one. :+1:

Index should be string `allow/0` to help you differentiate between default `int` composite index and your `str` custom composite index. Test `index` first too we don't want `int` and if it is an `int`, do a `return`.